src/pgrn-language-model-vectorize.c

```diff
@@ -0,0 +1,83 @@
+#include "pgroonga.h"
+
+#include "pgrn-groonga.h"
+#include "pgrn-language-model-vectorize.h"
+
+#include <catalog/pg_type_d.h>
+#include <utils/builtins.h>
+#ifdef PGRN_HAVE_VARATT_H
+#	include <varatt.h>
+#endif
+
+static grn_language_model_loader *loader = NULL;
+static grn_obj vector;
+
+PGDLLEXPORT PG_FUNCTION_INFO_V1(pgroonga_language_model_vectorize);
+
+void
+PGrnInitializeLanguageModelVectorize(void)
+{
+	loader = grn_language_model_loader_open(ctx);
+	GRN_FLOAT32_INIT(&vector, GRN_OBJ_VECTOR);
+}
+
+void
+PGrnFinalizeLanguageModelVectorize(void)
+{
+	GRN_OBJ_FIN(ctx, &vector);
+	if (loader)
+	{
+		grn_language_model_loader_close(ctx, loader);
+		loader = NULL;
+	}
+}
+
+Datum
+pgroonga_language_model_vectorize(PG_FUNCTION_ARGS)
+{
+	const char *tag = "[language-model-vectorize]";
+	text *modelName = PG_GETARG_TEXT_PP(0);
+
+	grn_language_model *model = NULL;
+	grn_language_model_inferencer *inferencer = NULL;
+
+	grn_language_model_loader_set_model(
+		ctx, loader, text_to_cstring(modelName), VARSIZE_ANY_EXHDR(modelName));
```

Comment:
We don't need `\0` terminated string here. It causes a needless copy.

```suggestion
		ctx, loader, VARDATA_ANY(modelName), VARSIZE_ANY_EXHDR(modelName));
```